ansible_collections/arista/cvp/plugins/module_utils/tag_tools.py

```diff
@@ -136,6 +138,13 @@ def tasker(self, tags: list, mode: string, auto_create: bool = True):
                         self.__cv_client.api.tag_config(element_type, workspace_id,
                                                         tag_name, tag_val)
                     if mode == 'delete':
+                        # unassign tags first
+                        self.__cv_client.api.tag_assignment_config(element_type,
+                                                                   workspace_id,
+                                                                   tag_name,
+                                                                   tag_val,
+                                                                   device_id,
+                                                                   "")
```

Comment:
I gave you the wrong API call. This is the call for `unassign` !! 
```suggestion
                        self.__cv_client.api.tag_assignment_config(element_type,
                                                                   workspace_id,
                                                                   tag_name,
                                                                   tag_val,
                                                                   device_id,                                                                
                                                                   "", remove=True)
```

Reply:
Fixed!